### PR TITLE
Support the pseudo-paths that git diff and friends will display

### PR DESCRIPTION
**Pull request by mannih**
Your looking at a commit in your git log and git/diff show two paths like
a/foo/bar and b/foo/bar. Simply copying one of those file names is difficult,
because simply double-clicking will include "a/" oder "b/".  So Open::This can
take a look at the path, see whether it starts with "a/" or "b/" and remove the
prefix if the rest of the path makes any sense.

### CODE REVIEW

**Comment by mannih** [comment]
Hmm, interesting. Why would only perl 5.30 think that my code wasn't tidy?

**Comment by oalders** [comment]
I only run author tests on that version of Perl, so it doesn't get picked up on the other versions.  If it's only an issue with tidying, I'm not concerned.  I can always run that after merge.

File: t/Open/This.t [hunk 1]
```diff
@@ -224,6 +224,24 @@ eq_or_diff(
...
+        'a git-diff path and the file exists'
+    );
+}
+
+eq_or_diff(
+    parse_text('a/t/test-data/i-m-not-here'),
+    undef,
+    'could have been a git diff file name, but it doesn\'t exist'
+);
+
```
Comment (oalders) [inline t/Open/This.t]: Should we add a variant of this test for `b` as well?

Comment (mannih) [reply t/Open/This.t]: Whatever you prefer. Should we?

Comment (oalders) [reply t/Open/This.t]: Please! :)

**Comment by mannih** [comment]
> I only run author tests on that version of Perl, so it doesn't get picked up on the other versions. If it's only an issue with tidying, I'm not concerned. I can always run that after merge.

I see. And now I get how tidyall does it's job. Just a difference in whitespace philosopies.

File: t/Open/This.t (hunk 1 above)
Comment (mannih) [reply t/Open/This.t]: I'm not doing this often enough. How can I ammend the test to the PR?

Comment (mannih) [reply t/Open/This.t]: Ah! github magic. Forget the question, please.

**Comment by oalders** [comment]
Thanks @mannih!

**Comment by oalders** [comment]
This change is now on its way to CPAN.  🎉

**Pull request merged**